components/exercises/chapter-4-oscar/DeepComparison.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { useState } from 'react';
+import { checkEqual, lastWeekList, currentWeekList, WeekItem } from './Utils';
+
+/**
+ * This represents an array of arrays of WeekItem objects, essentially creating a two-dimensional array of WeekItem objects. Each element of the outer array is an array of WeekItem objects.
+ */
+type GroceryList = WeekItem[][];
+
+const newElement: WeekItem = {
+  name: '🍅 Tomatoes',
+  description: '1kg Fresh tomatoes',
+  price: 0.5,
+};
+
+/**
+ * Renders a component that compares two grocery lists and allows adding/removing items.
```

Comment:
and checks for deep equality of both grocery lists

Reply:
added